### ISSUE TITLE
sdist has world-writable permissions

### ISSUE THREAD

**Issue by minrk** [urllib3/urllib3#1415]
The sdist for urllib3 1.23 on PyPI has some surprising world-writable permissions on a number of files:

```
 root@6821c0c367c3:~/urllib3-1.23# ls -lan
total 152
drwxrwxrwx 8 1000 1000  4096 Jul 19 07:56 .
drwx------ 1    0    0  4096 Jul 19 07:56 ..
-rwxrwxrwx 1 1000 1000 28535 Jun  5 03:16 CHANGES.rst
-rwxrwxrwx 1 1000 1000  8542 Jun  5 03:16 CONTRIBUTORS.txt
-rwxrwxrwx 1 1000 1000  1175 Jun  5 03:04 LICENSE.txt
-rwxrwxrwx 1 1000 1000   204 Jun  5 03:04 MANIFEST.in
-rwxrwxrwx 1 1000 1000  1192 Jun  5 03:16 Makefile
-rwxrwxrwx 1 1000 1000 41888 Jun  5 03:22 PKG-INFO
-rwxrwxrwx 1 1000 1000  3780 Jun  5 03:16 README.rst
drwxr-xr-x 3    0    0  4096 Jul 19 07:56 build
-rwxrwxrwx 1 1000 1000   212 Jun  5 03:16 dev-requirements.txt
drwxrwxrwx 4 1000 1000  4096 Jun  5 03:22 docs
drwxrwxrwx 3 1000 1000  4096 Jun  5 03:22 dummyserver
-rw-r--r-- 1    0    0  5966 Jul 19 07:56 record.txt
-rwxrwxrwx 1 1000 1000   620 Jun  5 03:22 setup.cfg
-rwxrwxrwx 1 1000 1000  2899 Jun  5 03:16 setup.py
drwxrwxrwx 5 1000 1000  4096 Jun  5 03:22 test
drwxrwxrwx 5 1000 1000  4096 Jun  5 03:22 urllib3
drwxrwxrwx 2 1000 1000  4096 Jun  5 03:22 urllib3.egg-info
```

Installing this with the old `python setup.py install --single-version-externally-managed` command results in world-writable metadata files being installed with the package (in .egg-info). Installing with pip does not appear to have this issue.

I ran the release.sh script on my system and got much more expected permissions:

```
$ ls -la urllib3-dev
total 140
drwxr-xr-x 6 root root  4096 Jul 19 08:03 .
drwxr-xr-x 3 root root  4096 Jul 19 08:03 ..
-rw-r--r-- 1 root root 28841 Jul 19 08:03 CHANGES.rst
-rw-r--r-- 1 root root  8667 Jul 19 08:03 CONTRIBUTORS.txt
-rw-r--r-- 1 root root  1175 Jul 19 08:03 LICENSE.txt
-rw-r--r-- 1 root root   204 Jul 19 08:03 MANIFEST.in
-rw-r--r-- 1 root root  1192 Jul 19 08:03 Makefile
-rw-r--r-- 1 root root 42281 Jul 19 08:03 PKG-INFO
-rw-r--r-- 1 root root  3780 Jul 19 08:03 README.rst
-rw-r--r-- 1 root root   221 Jul 19 08:03 dev-requirements.txt
drwxr-xr-x 4 root root  4096 Jul 19 08:03 docs
drwxr-xr-x 3 root root  4096 Jul 19 08:03 dummyserver
-rw-r--r-- 1 root root   620 Jul 19 08:03 setup.cfg
-rwxr-xr-x 1 root root  2937 Jul 19 08:03 setup.py
drwxr-xr-x 4 root root  4096 Jul 19 08:03 src
drwxr-xr-x 5 root root  4096 Jul 19 08:03 test
```

so I'm a bit baffled as to how it ended up that way. It would be nice for the next sdist to have more normal permissions in the tarball. Maybe it's some configuration on the system where the release script is typically run?

**Comment by SethMichaelLarson** [comment]
I'll check this out, we definitely should ensure that world-writeable is not the default for our sdist in our new release protocol.

**Comment by crwilcox** [comment]
@sethmlarson is this resolved now or will be by your modifications to releasing (https://github.com/urllib3/urllib3/pull/1508)?

**Comment by sethmlarson** [comment]
This will in theory fix that issue by removing the human element of releases unless there's some special transformation we need to do to the tarball after building it. Does the default sdist build have world-writable perms?

**Comment by sethmlarson** [comment]
Closed via #1508
